cmd/kubeadm/app/util/kustomize/kustomize.go

```diff
@@ -145,33 +201,45 @@ func (km *Manager) Kustomize(res []byte) ([]byte, error) {
 		fakeDir = `C:\`
 	}
 
-	// write resources and patches to the in memory fs, generate the kustomization.yaml
-	// that ties everything together
-	kustomization.WriteString("resources:\n")
-	for i, r := range resources {
-		b, err := r.MarshalJSON()
+	// writes the resource to a file in the temp file system
+	b, err := yaml2.Marshal(resource)
+	if err != nil {
+		return nil, err
+	}
+	name := "resource.yaml"
+	_ = memFS.WriteFile(filepath.Join(fakeDir, name), b)
+
+	km.k.Resources = []string{name}
+
+	// writes strategic merge patches to files in the temp file system
+	km.k.PatchesStrategicMerge = []patch.StrategicMerge{}
+	for i, p := range strategicMerge {
+		b, err := yaml2.Marshal(p)
 		if err != nil {
 			return nil, err
 		}
-
-		name := fmt.Sprintf("resource-%d.json", i)
+		name := fmt.Sprintf("patch-%d.yaml", i)
 		_ = memFS.WriteFile(filepath.Join(fakeDir, name), b)
 		fmt.Fprintf(&kustomization, " - %s\n", name)
```

Comment:
This line needs to be removed along with `var kustomization bytes.Buffer` above.